lib/state.js

```diff
@@ -569,6 +569,82 @@ class State {
 
     return result.rowCount;
   }
+   
+   /**
+   * Insert an AMI's first usage.
+   */
+  async insertAmiUsage({region, id}, client) {
+    assert(typeof region === 'string');
+    assert(typeof id === 'string');
+
+    let text = [
+      "INSERT INTO amiusage (region, id, lastused)",
+      "VALUES ($1, $2, CURRENT_TIMESTAMP)"
+    ].join(' ');
+    let values = [region, id];
+    let name = 'insert-amiusage';
+
+    let result;
+    if (client) {
+      result = await client.query({text, values, name});
+    } else {
+      result = await this._pgpool.query({text, values, name});
+    }
+    assert(result.rowCount === 1, 'inserting AMI usage had incorrect rowCount');
+  }
+   
+   /**
+   * Either insert or update an AMI's usage.  This ensures an atomic insert or
+   * update, but we never learn which one happened. Only updates the row 
+   * if the proposed time is newer than the existing time.
+   */
+  async upsertAmiUsage({region, id}, client) {
+    assert(typeof region === 'string');
+    assert(typeof id === 'string');
+    
+    let text = [
```

Comment:
To be more consistent with the other upserts, could we break the query up like this:
```javascript
    let text = [
      "INSERT INTO amiusage (region, id, lastused)",
      "VALUES ($1, $2, CURRENT_TIMESTAMP)",
      "ON CONFLICT (region, id) DO UPDATE SET lastused = CURRENT_TIMESTAMP"
    ].join(' ');
```

Reply:
I would also go with template strings.